include/tscpp/api/HttpMsgComp.h

```diff
@@ -0,0 +1,424 @@
+/**
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+/**
+ * @file HttpMsgComp.h
+ *
+ * Classes for convenient manipulation of HTTP message components.
+ */
+
+/*
+Note:  The clases in this include file may be used independently from the rest of the C++ API.
+*/
+
+#pragma once
+
+// Suppress assert() macro in non-debug builds.
+#if !defined(NDEBUG) && !defined(DEBUG)
+#define NDEBUG
+#endif
+#include <cassert>
+
+#include <string_view>
+
+#include <ts/ts.h>
+
+#include <tscpp/api/noncopyable.h>
```

Comment:
Why bother with this?

Reply:
You mean just delete copy constructor / assign instead?

Reply:
Yes. There have been multiple passes on the code base to make exactly that change and get rid of the "noncopyable" stuff.